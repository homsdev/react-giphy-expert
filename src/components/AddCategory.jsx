import { useState } from "react";


export const AddCategory = ({ onNewCategory }) => {

    const [inputValue, setInputValue] = useState('One Punch');

    const onInputChange = ({ target }) => {
        setInputValue(target.value);
    }

    /*const onAddCategory = (event) => {
        event.preventDefault();
        if (inputValue.trim() <= 1) return;
        setCategories((categories) => [...categories, inputValue]);
        setInputValue('');
    }*/

    const onSubmit = (event) => {
        event.preventDefault();
        if (inputValue.trim() <= 1) return;
        onNewCategory(inputValue.trim());
        setInputValue('');
    }


    return (
        <form onSubmit={onSubmit}>
            <input type="text"
                placeholder="Search Gif"
                value={inputValue}
                onChange={onInputChange}
            />
        </form>
    )
}

