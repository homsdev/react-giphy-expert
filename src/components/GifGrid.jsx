import { useFetchGifs } from "../hooks/useFetchGifs";
import { GifItem } from "./GifItem";
import PropTypes from "prop-types";

export const GifGrid = ({ category }) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && <p>Loading...</p>
            }
            <div className="card-grid" >
                {images.map((image) => (
                    <GifItem
                        key={image.id}
                        {...image} /> /**Valid when there are a lot of properties */
                ))}
            </div>
        </>
    )
}

GifGrid.propTypes = {
    category: PropTypes.string.isRequired
}


