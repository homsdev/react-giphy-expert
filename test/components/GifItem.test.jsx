import { render, screen } from "@testing-library/react"
import { GifItem } from "../../src/components/GifItem"

describe('Test on <GifItem.jsx />', () => {

    const title = 'Death Note';
    const url = 'https://www.deathnote.js/';

    test('should match with snapshot', () => {
        const { container } = render(<GifItem title={title} url={url} />);
        expect(container).toMatchSnapshot();
    });

    test('should show image with received ALT and URL', () => {
        render(<GifItem title={title} url={url} />)
        //screen.debug();
        //expect(screen.getByRole('img').src).toBe(url);
        //expect(screen.getByRole('img').alt).toBe(title);

        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    })

    test('should show gif title', () => {
        render(<GifItem title={title} url={url} />)
        expect(screen.getByText(title)).toBeTruthy();
    })

})
