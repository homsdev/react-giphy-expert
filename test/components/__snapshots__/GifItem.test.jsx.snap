// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on <GifItem.jsx /> should match with snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Death Note"
      src="https://www.deathnote.js/"
    />
    <p>
      Death Note
    </p>
  </div>
</div>
`;
